dotnet/samples/Concepts/Agents/ComplexChat_NestedShopper.cs

```diff
@@ -0,0 +1,227 @@
+        ChatCompletionAgent agentReviewer = CreateAgent(ReviewerName, ReviewerInstructions);
+
+        KernelFunction innerSelectionFunction = KernelFunctionFactory.CreateFromPrompt(InnerSelectionInstructions, jsonSettings);
+        KernelFunction outerTerminationFunction = KernelFunctionFactory.CreateFromPrompt(OuterTerminationInstructions, jsonSettings);
+
+        AggregatorAgent agentShopperGroup =
+            new(CreateChat)
+            {
+                Name = "Shopper",
+                Mode = AggregatorMode.Nested,
+            };
+
+        AgentGroupChat chat =
+            new(agentShopperGroup)
+            {
+                ExecutionSettings =
+                    new()
+                    {
+                        TerminationStrategy =
+                            new KernelFunctionTerminationStrategy(outerTerminationFunction, CreateKernelWithChatCompletion())
+                            {
+                                ResultParser =
+                                    (result) =>
+                                    {
+                                        OuterTerminationResult? jsonResult = JsonResultTranslator.Translate<OuterTerminationResult>(result.GetValue<string>());
+
+                                        return jsonResult?.isAnswered ?? false;
+                                    },
+                                MaximumIterations = 5,
+                            },
+                    }
+            };
+
+        // Invoke chat and display messages.
+        this.WriteLine("\n######################################");
+        this.WriteLine("# DYNAMIC CHAT");
+        this.WriteLine("######################################");
+
+        await InvokeChatAsync("Can you provide three original birthday gift ideas.  I don't want a gift that someone else will also pick.");
+
+        await InvokeChatAsync("The gift is for my adult brother.");
+
+        if (!chat.IsComplete)
+        {
+            await InvokeChatAsync("He likes photography.");
+        }
+
+        this.WriteLine("\n\n######################################");
+        this.WriteLine("# AGGREGATED CHAT");
+        this.WriteLine("######################################");
+
+        await foreach (var content in chat.GetChatMessagesAsync(agentShopperGroup).Reverse())
+        {
+            this.WriteLine($"# {content.Role} - {content.AuthorName ?? "*"}: '{content.Content}'");
+        }
+
+        async Task InvokeChatAsync(string input)
+        {
+            chat.AddChatMessage(new ChatMessageContent(AuthorRole.User, input));
+
+            this.WriteLine($"# {AuthorRole.User}: '{input}'");
+
+            await foreach (var content in chat.InvokeAsync(agentShopperGroup))
+            {
+                this.WriteLine($"# {content.Role} - {content.AuthorName ?? "*"}: '{content.Content}'");
+            }
+
+            this.WriteLine($"\n# IS COMPLETE: {chat.IsComplete}");
+        }
```

Comment:
When printing the inner chat can we indent this and/or add a prefix `>>>> ` so it's clear which are internal versus external messages. In the PR comment you need to read carefully to see this.